### ISSUE TITLE
Multiple .Filter calls do not work together

### ISSUE THREAD

**Issue by nirinchev** [object/Simple.OData.Client#78]
Currently, only the first Filter gets evaluated and everything after that gets disregarded. It would be better if we could chain multiple filters like the .Where LINQ syntax.
For example, with Simple.OData.Client the following doesn't work:
```client.For<Foo>().Filter(f => f.Name == "Bar").Filter(f => f.Age > 15)```

**Comment by object** [comment]
You can evaluate multiple filter expressions, just use && or || operators,e.g.:

client.For<Foo>().Filter(f => f.Name == "Bar" && f => f.Age > 15)

Hope this helps.

**Comment by nirinchev** [comment]
Yes, I know I can do this in a single .Filter call, but I believe support for chaining multiple .Filter calls would be much appreciated. This would allow manipulating the query in different places in code. For example I can add a date filter in one method and then pass the query to another method to add another filter. It would also allow mixing fluent and basic filters as a workaround for unsupported functions. For example, currently Geography.Distance seems to be unsupported, so I can build my string filter and add it to my fluent api filtered query like:

```
var distanceFilter = string.Format ("geo.distance(Location,geography'SRID=4326;POINT ({0} {1})') lt {2}",
    location.Longitude,
    location.Latitude,
    distance);

var items = await client.For<School>().Filter(s => s.StudentCount > 10).Filter(distanceFilter).FindEntriesAsync();
```

**Comment by object** [comment]
I see your point. I have to think how this can be achieved without sacrificing API consistency. If Filter clause can be chained, then what about Expand, Select and OrderBy? They should probably also support chaining. So I am not quite sure this feature will naturally fit, but let me think about it.

**Comment by object** [comment]
Hi, your suggestion hasn't been forgotten. It will be a part of forthcoming 4.5 release which will be out sometime next week. In the meantime you can check out its pre-release from MyGet feed:

https://www.myget.org/feed/Packages/simple-odata-client

**Comment by object** [comment]
Implemented in 4.5, available from MyGet feed until the release: https://www.myget.org/feed/Packages/simple-odata-client
